### DISCUSSION TITLE
How to change directory of downloaded models

### DISCUSSION THREAD

**Discussion by somecoolguy1397** [idiap/coqui-ai-TTS#215] [Q&A] [answered]
The original TTS seems to have added this, but it isn't added here.
In the original TTS, you change or add an environment variable called "TTS_HOME" or something like that. This version uses trainer.io instead of the library itself.

**Comment by somecoolguy1397** [comment]
I've made a temporary fix on my PC, replacing
```
from trainer.io import get_user_data_dir
```
with
```
from trainer.io import get_user_data_dir as trainer_gudd
def get_user_data_dir(_model):
    if _model == "tts":
        return "C:\\path\\to\\dir"
    else:
        return trainer_gudd(_model)
```
,and it seems to work.

**Reply by eginhard** [reply, marked as answer]
You can still set the TTS_HOME or XDG_DATA_HOME environment variables to set a custom folder, that hasn't changed. See https://coqui-tts.readthedocs.io/en/latest/faq.html#where-does-coqui-store-downloaded-models
